src/rocksdb.rs

```diff
@@ -83,7 +83,8 @@ impl <'a> Iterator for SubDBIterator<'a> {
             let key = unsafe { slice::from_raw_parts(key_ptr, key_len as usize) };
             let val_ptr = unsafe { rocksdb_ffi::rocksdb_iter_value(native_iter, val_len_ptr) };
             let val = unsafe { slice::from_raw_parts(val_ptr, val_len as usize) };
-            Some((key,val))
+
+            Some((key.to_vec().into_boxed_slice(),val.to_vec().into_boxed_slice()))
```

Comment:
Ahh I see, makes sense.  One thing I may like eventually is to have a codec that is tied to a particular column family, rather than DB-wide, so feel free to implement it DB-wide in the mean time but with the knowledge that we may want to make it CF-specific somehow down the road.